docs/core/extensions/logger-message-generator.md

```diff
@@ -0,0 +1,320 @@
+{
+    [LoggerMessage(EventId = 0, Level = LogLevel.Critical, Message = "Could not open socket to `{hostName}`")]
+    public static partial void CouldNotOpenSocket(ILogger logger, string hostName);
+}
+```
+
+The above sample shows the canonical use of the logging source generator, where the logging method is static and the log level is specified in the attribute definition. There are other forms possible too. For example, in the following example the logging method is declared as an instance method. In this form, the logging method gets the logger by accessing an `ILogger` field in the containing class.
+
+```csharp
+class MyLogWrapper
+{
+    private readonly ILogger _logger;
+
+    public MyLogWrapper(ILogger logger)
+    {
+        _logger = logger;
+    }
+
+    [LoggerMessage(EventId = 0, Level = LogLevel.Critical, Message = "Could not open socket to `{hostName}`")]
+    public partial void CouldNotOpenSocket(string hostName);
+}
+
+```
+
+Sometimes, the logging level needs to be a dynamic property rather than being statically built into the code. You can do this by omitting the logging level from the attribute and instead supplying it as an argument to the logging method:
+
+```csharp
+static partial class Log
+{
+    [LoggerMessage(EventId = 0, Message = "Could not open socket to `{hostName}`")]
+    public static partial void CouldNotOpenSocket(ILogger logger, LogLevel level, string hostName);
+}
+```
+
+You can omit the logging message and an empty string will be provided for the message but the state will contain the arguments and formats them as key value pairs.
+
+```csharp
+using Microsoft.Extensions.Logging;
+using System.Text.Json;
+
+namespace LoggingGeneratorSampleApp
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            var logger = LoggerFactory.Create(builder =>
+            {
+                builder.AddJsonConsole(o => 
+                o.JsonWriterOptions = new JsonWriterOptions()
+                {
+                    Indented = true
+                });
+            }).CreateLogger<Program>();
+
+            logger.LogMethod(LogLevel.Information, "Liana", "California");
+        }
+    }
+
+    public static partial class MyPoco
+    {
+        [LoggerMessage(EventId = 23)]
+        public static partial void LogMethod(this ILogger logger, LogLevel logLevel, string name, string state);
+    }
+}
+```
+
+Outputs:
+```
+{
+  "EventId": 23,
+  "LogLevel": "Information",
+  "Category": "ConsoleApp66.Program",
+  "Message": "",
+  "State": {
+    "Message": "",
+    "name": "Liana",
+    "state": "California",
+    "{OriginalFormat}": ""
+  }
+}
+```
+
+## Requirements
+
+Logging methods have some constraints which must be followed:
+
+- Logging methods must be static, partial, and return void.
+- Logging method names must not start with an underscore.
+- Parameter names of logging methods must not start with an underscore.
+- Logging methods may not be defined in nested type.
```

Comment:
If https://github.com/dotnet/runtime/issues/52301 should be addressed, this may be no longer true.